applications/GeoMechanicsApplication/custom_elements/triangle_2d_15.hpp

```diff
@@ -0,0 +1,1753 @@
+            rResult(2, 1) = (128.0 * y3 - 144.0 * y2 + 44.0 * y1 - 3.0) / 3.0;
+
+            rResult(3, 0) = -128.0 * (tC2 - 0.5 * tC1 + 1.0 / 24.0) * x1 + (128.0 * tC3 - 96.0 * tC2 + 16.0 * tC1) / 3.0;
+            rResult(3, 1) = -16.0 * x1 * (24.0 * tC2 - 12.0 * tC1 + 1.0) / 3.0;
+
+            rResult(4, 0) = -128.0 * (x1 - 0.25) * (tC1 - 0.125) * x1 + 128.0 * (x1 - 0.125) * (tC1 - 0.25) * tC1;
+            rResult(4, 1) = -4.0 * x1 * (4.0 * x1 - 1.0) * (8.0 * tC1 - 1.0);
+
+            rResult(5, 0) = -(128.0 * x3 - 96.0 * x2 + 16.0 * x1) / 3.0 + 128.0 * (x2 - 0.5 * x1 + 1.0 / 24.0) * tC1;
+            rResult(5, 1) = -16.0 * x1 * (8.0 * x2 - 6.0 * x1 + 1.0) / 3.0;
+
+            rResult(6, 0) = 16.0 * y1 * (24.0 * x2 - 12.0 * x1 + 1.0) / 3.0;
+            rResult(6, 1) = (128.0 * x3 - 96.0 * x2 + 16.0 * x1) / 3.0;
+
+            rResult(7, 0) = 4.0 * (8.0 * x1 - 1.0) * (4.0 * y1 - 1.0) * y1;
+            rResult(7, 1) = 4.0 * (4.0 * x1 - 1.0) * (8.0 * y1 - 1.0) * x1;
+
+            rResult(8, 0) = (128.0 * y3 - 96.0 * y2 + 16.0 * y1) / 3.0;
+            rResult(8, 1) = 16.0 * x1 * (24.0 * y2 - 12.0 * y1 + 1.0) / 3.0;
+
+            rResult(9, 0) = -16.0 * y1 * (8.0 * y2 - 6.0 * y1 + 1.0) / 3.0;
+            rResult(9, 1) = -(128.0 * y3 - 96.0 * y2 + 16.0 * y1) / 3.0 + 128.0 * (y2 - 0.5 * y1 + 1.0 / 24.0) * tC1;
+
+            rResult(10, 0) = -4.0 * y1 * (4.0 * y1 - 1.0) * (8.0 * tC1 - 1.0);
+            rResult(10, 1) = -128.0 * (y1 - 0.25) * (tC1 - 0.125) * y1 + 128.0 * (y1 - 0.125) * tC1 * (tC1 - 0.25);
+
+            rResult(11, 0) = -16.0 * y1 * (24.0 * tC2 - 12.0 * tC1 + 1.0) / 3.0;
+            rResult(11, 1) = -128.0 * (tC2 - 0.5 * tC1 + 1.0 / 24.0) * y1 + (128.0 * tC3 - 96.0 * tC2 + 16.0 * tC1) / 3.0;
+
+            rResult(12, 0) = 256.0 * y1 * (-x1 * (tC1 - 0.125) + 0.5 * tC2 - 0.125 * tC1);
+            rResult(12, 1) = 256.0 * x1 * (-y1 * (tC1 - 0.125) + 0.5 * tC2 - 0.125 * tC1);
+
+            rResult(13, 0) = -32.0 * y1 * (4.0 * x2 - x1) + 256.0 * (x1 - 0.125) * y1 * tC1;
+            rResult(13, 1) = 128.0 * (x1 - 0.25) * (-y1 + tC1) * x1;
+
+            rResult(14, 0) = 128.0 * (y1 - 0.25) * y1 * (-x1 + tC1);
+            rResult(14, 1) = -32.0 * x1 * (4.0 * y2 - y1) + 256.0 * (y1 - 0.125) * tC1 * x1;
+            //
+            return rResult;
+        }
+
+
+        // ========================================================================================
+        // ========================================================================================
+        /**
+         * returns the shape function gradients in an arbitrary point,
+         * given in local coordinates
+         *
+         * @param rResult the matrix of gradients,
+         * will be overwritten with the gradients for all
+         * shape functions in given point
+         * @param rPoint the given point the gradients are calculated in
+         */
+        virtual Matrix& ShapeFunctionsGradients(Matrix& rResult, CoordinatesArrayType& rPoint)
```

Comment:
shouldn't be "const CoordinatesArrayType& rPoint"?

Reply:
Done